# ml R/3.5.1-foss-2016b-BABS

library(DESeq2)
library(biomaRt)
library(ggplot2)
library("RColorBrewer")
library("gplots")
library("genefilter")
library("lattice")
library("pheatmap")
library(tidyr)
library(dplyr)
library(d3heatmap)
library(plotly)
library(ggrepel)
library(htmlwidgets)
library(pcaExplorer)


projectDir <- "/camp/stp/babs/working/sopenam/projects/"
### Change the following entries for each project accordingly
### change the following entries and sample names ###

PI <- "tybulewiczv/"
scientist <- "eva.lana-elola/"
project <- "12_RNASeq_HumanEmbryonicHearts_DownSyndrome/"

#directory where to export results (NEW DIRECTORY TO AVOID OVERWRITTING DATA)
DESeq.Dir <- paste0(projectDir,PI, scientist, project, "/results/DESeq/")

######### GET COUNTS FROM RSEM   ##################################
# this scripts reads count tables generated by rsem,
# it requires a file called design.csv, that contains the experimental design
# you need to know the number of rows to import (use wc -l on terminal) and replace the value in line10

# get the file names
files <- dir(paste0(projectDir, PI, scientist, project, "/data/align/rsem/"),pattern=".genes.results", recursive=T)
#create a df, where the data read from the files is added
nfiles=length(files)
genes=58233 # obtained using wc-l in linux and substracting 1 (header)

data<- matrix(ncol=nfiles, nrow=genes)
# nrow should be obtained from the annotation file or one of the counting files
# ncol should be the number of samples
i=1
names<-c()
#loop for the analysis
if (length(files) == 0) {
  stop("No files found!")
} else {
  for (file in files) {
    print(file)
    # extract name from input file
    temp <- strsplit(file, "[/]")
    temp <- temp[[1]]
    
    a <- read.table( paste0(projectDir, PI, scientist, project, "/data/align/rsem/",file), sep="\t", header=T, stringsAsFactors=F)
    #extract expected_count
    counts <- a$expected_count
    counts <- as.matrix(round(counts))
    names <- c(names,temp[1])
    #data<- cbind(data, counts)
    data[,i]<- counts
    i=i+1
  }
}

countData <- as.data.frame(data)
names(countData) <- names
rownames(countData) <- a$gene_id 
countData<- as.matrix(countData)

## Samples to be removed A19 because it has very low reads and library quality wasn't good
## For Sample 8, we were supposed to use MSC and by mistake we have used CD34+ cells. So, this sample is useless on its own.
## For Sample 49, we can discard the data for now for this sample.

## may have to remove A26, A28, A40
# countData <- countData[, - which(colnames(countData) == "LAN757A19")] 

write.table( countData, paste0( DESeq.Dir, "Table_counts_per_gene_per_Sample_RN20085.txt"), sep="\t" , col.names=NA, quote= FALSE)

### Get TPM values, so I'm extracting that data for him as well
tpmdata<- matrix(ncol=nfiles, nrow=genes)
# nrow should be obtained from the annotation file or one of the counting files
# ncol should be the number of samples
i=1
names<-c()
#loop for the analysis
if (length(files) == 0) {
  stop("No files found!")
} else {
  for (file in files) {
    print(file)
    # extract name from input file
    temp <- strsplit(file, "[/]")
    temp <- temp[[1]]
    
    a <- read.table( paste0(projectDir, PI, scientist, project, "/data/align/rsem/",file), sep="\t", header=T, stringsAsFactors=F)
    #extract TPM
    counts <- a$TPM
    counts <- as.matrix(counts)
    names <- c(names,temp[1])
    #data<- cbind(data, counts)
    tpmdata[,i]<- counts
    i=i+1
  }
}

tpmdata <- as.data.frame(tpmdata)
names(tpmdata) <- names
rownames(tpmdata) <- a$gene_id 

 #tpmdata <- tpmdata[, - which(colnames(tpmdata) == "LAN757A19")] 

write.table(tpmdata, file=paste0(DESeq.Dir,"TPM_table_all_ENSGenes.txt"),
            sep="\t", col.names=NA, quote=F)

## prepare sample table

colData <- read.csv( paste0(projectDir, PI, scientist, project, "docs/design.csv"),
                       header=T, stringsAsFactors = F)

#colData <- colData[, - which(colData$sample == "LAN757A19")] 

colData$treatment <- factor(colData$treatment)
colData$Gender <- factor(colData$Gender)

# Give rownames for colData
rownames(colData) <- colData$sample

#colData <- colData[ - which(rownames(colData) == "LAN757A19"),]

# Re-order colData based on the column names of countData
colData <- colData[colnames(countData),]
# It's always worth checking the following is always TRUE (and none of the values are NA or FALSE) prior to creating the dds object.
colnames(countData) == rownames(colData)


## Experiment aims to investigate the DE genes between MDS and healthy at different time points
dds <- DESeqDataSetFromMatrix(countData = countData,
                              colData = colData,
                              design = ~ Gender + treatment)

#run DESeq 
dds <- DESeq(dds, betaPrior = F)
## because there are several time points and we don't want all the comparisons I'll test the betaPrior option

#######################################################################################
### Data quality asessment, Data transformations and visualization
#######################################################################################

sf<-sizeFactors(dds)
names(sf)<- colData$group
#jpeg(file=paste(DESeq.Dir,"Plot_size_factors.jpeg",sep=""))
barplot(sf, las=2, cex.names=0.75)
#dev.off()

rld1 <- rlogTransformation(dds, blind=TRUE)


##box plots of the data
##box plots of the data show median around 0
boxplot(assay(rld1), names=as.character(colData[colnames(rld1),6]), cex.axis=0.6, las=2, outline=F, na.action=na.omit)
## median around 0 !!!
rld1.bplot <- assay(rld1[! rowSums(assay(rld1)) == 0,])
#jpeg(file=paste(DESeq.Dir,"Boxplot_norm_counts.jpeg",sep=""), width = 680, height = 680)
#par(mar=c(12,5,10,5))
boxplot(rld1.bplot, names=as.character(colData[colnames(rld1),1]), cex.axis=0.6, las=2, outline=F, na.action=na.omit)
#dev.off()
# save box plot with 
#jpeg(file=paste(DESeq.Dir,"Boxplot_norm_counts.names.jpeg",sep=""), width = 680, height = 680)
#par(mar=c(12,5,10,5))
boxplot(rld1.bplot, names=rownames( colData), cex.axis=0.6, las=2, outline=F, na.action=na.omit)
#dev.off()

### PCA plots ### 

library("RColorBrewer")
datacustomPCA <- plotPCA(rld1,intgroup= c("treatment", 
                                          "Gender", 
                                          "sample"), returnData=TRUE)
percentVar <- round(100 * attr(datacustomPCA, "percentVar"))

ggplot(datacustomPCA, aes(PC1, PC2, color= treatment, shape= Gender)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance")) +
  coord_fixed() + 
  theme(text=element_text(size=7) )+
  #scale_color_brewer(palette="Paired")+ 
  geom_text_repel(aes(label = sample),
                  segment.alpha = 0.5 )
ggsave(paste0(DESeq.Dir,"PCAplot.pdf") )

## Heatmap of the sample-to-sample distances
library("RColorBrewer")
library("pheatmap")

distsRL1 <- dist(t(assay(rld1)))

sampleDistMatrix <- as.matrix(distsRL1)
rownames(sampleDistMatrix) <- colData$Sample.ReplicateGroup
colnames(sampleDistMatrix) <- NULL
colors <- colorRampPalette(rev(brewer.pal(9,"Blues"))) (255)

jpeg(file=paste0(DESeq.Dir,"Heatmap_sampleTosample_distance.condition.jpeg"), width = 1024, height = 768)
pheatmap (sampleDistMatrix, 
          clustering_distance_rows=distsRL1,
          clustering_distance_cols=distsRL1,
          col=colors)
dev.off()

#####################################################################################################
########################################## DESeq Results ############################################
#####################################################################################################

#extract results  DESeq Object
resultsNames(dds)
# 1
res_DS_vs_control <- results(dds, contrast=c("treatment", "DS", "control"))

##### using bioMart   ##############
# extract gene symbols to match the ENSEMBLGENE IDs
lookupgenes <- names(dds)

ensembl = useMart ( biomart="ENSEMBL_MART_ENSEMBL", host="may2017.archive.ensembl.org")
human = useDataset("hsapiens_gene_ensembl", mart =ensembl)

#listAttributes(ensembl85)
lookup<-getBM(attributes = c("ensembl_gene_id", "external_gene_name", "chromosome_name", "start_position", "end_position"), filters = "ensembl_gene_id", values = lookupgenes, mart = human)
################     ##################

#####################################################
#####      generate  MA plot for the data      #####
#####################################################
results_list <-ls(pattern="^res_") 
results_list
sigdifgenes <- c()

for (f in results_list) {
  # read each results list
  print(f)
  i<- assign(f, get(f))
  #################################################
  ###### print MA plots for DESeq results  ###### 
  #################################################
  
  jpeg(file=paste(DESeq.Dir,f,"_MA_plot.jpg",sep=""))
  #pdf(file=paste(DESeq.Dir,f,"_MA_plot.pdf",sep=""))
  DESeq2::plotMA (i, alpha=0.05, ylim=c(-10,12),  main=paste("log2 fold change",f,"padj<0.05",sep="_"))
  abline(h=1)
  abline(h=-1)
  dev.off()
  ######################################################
  #####  extract statistically significant genes #####
  ######################################################
  y <- na.omit (i)
  y <- y[ y$padj < 0.05, ] #extract significant values padj<0.05
  y <- as.data.frame(y) #convert DESeq object into data frame
  assign(gsub("res_", "DF_",f), y )
  sigdifgenes <- c(sigdifgenes, rownames(y))
  
  #############################################################################
  ##print results to text file (total DE genes, up and donwregulated)#######
  #############################################################################
 # sink(paste0(DESeq.Dir,"summary_results_DESeq_analysis.txt"), append =T)
 # cat("Total number DE genes: ");cat(nrow(y));cat("\n");
 # cat("number of UP and DOWN regulated genes: ");cat(table(y$log2FoldChange < 0));cat("\n");
 # cat("number of UP log2FC>1: ");cat(length(which(y$log2FoldChange > 1)));cat("\n");
 # cat("number of DOWN log2FC<-1: ");cat(length(which(y$log2FoldChange < -1)));cat("\n");
 # cat("\n");
 # sink()
  # add gene names
  y <- merge(y, lookup, by.x=0, by.y=1, all.x=T, all.y=F)
  # order by highest DE gene
  y <- y[order(y$log2FoldChange, decreasing=T),]
  # save the data frame
  # save(y, file=paste(DESeq.Dir,f, ".RData", sep=""))
  assign(gsub("res_", "DF_",f), y )
  write.table( y, file= paste (DESeq.Dir,f,"_results.txt", sep=""), row.names=F, quote=F, sep="\t")
}

## sigdifgenes contains the names of all DE GENES,but some are repeated, 
sigdifgenes <- unique(sigdifgenes) # get list of DE genes not repeated


######################################################################
#### ######  write results as rnk file for GSEA Analysis #### ###### 
######################################################################
for (f in results_list) {
  # read each results list
  print(f)
  i<- assign(f, get(f))
  ### dir to save data to
  GSEA.Dir <- paste0(projectDir, PI, scientist, project, "results/GSEA/")
  x <- na.omit (i) # remove N/A
  gsea.data <- as.data.frame(x)
  gsea.data$gene_id <- row.names(gsea.data)
  gsea.data <- gsea.data[,c(7,4)]
  gsea.data <- gsea.data[order(gsea.data$stat, decreasing = T),]
  assign(paste0("rnk_",f), gsea.data)
  write.table(gsea.data, file= paste (GSEA.Dir,f,".rnk", sep=""), row.names=F, quote=F, sep="\t")
}
####### Diagnostic plots (https://www.bioconductor.org/help/course-materials/2015/LearnBioconductorFeb2015/B02.1.1_RNASeqLab.html#de)

### Get DEGS for all genes in Chr21
lookup$chr_coord <- paste( lookup$chromosome_name, paste(lookup$start_position, lookup$end_position, sep="-"), sep=":")
genes_in_chr21 <- merge( as.data.frame(res_DS_vs_control), lookup, by.x=0, by.y=1)
genes_in_chr21 <- genes_in_chr21[genes_in_chr21$chromosome_name == "21",c(1:8,12) ]
genes_in_chr21 <- genes_in_chr21[ order( genes_in_chr21$chr_coord, decreasing = F),]

######### GENE CLUSTERING ############################
library("genefilter")
#### function ####

distEisen <- function(x, use = "pairwise.complete.obs") {
  co.x <- cor(x, use = use)
  dist.co.x <- 1 - co.x
  return(as.dist(dist.co.x))
}


#### heatmap for list of genes sigdiffgenes ####

colors <- colorRampPalette( rev(brewer.pal(9, "PuOr")) )(255)

sidecols <-  brewer.pal( 8,"Set1")[ rld1$treatment]
#sidecols <- cbp[seq(1,24,by=3)][ rld1$condition ]
mat1 <- assay(rld1)[ rownames(assay(rld1)) %in% sigdifgenes, ]
colnames(mat1) <- rld1$condition
Rowv   <- as.dendrogram(hclust(distEisen(t(mat1)), method = "ave"));

jpeg(file.path(DESeq.Dir,"Heatmap_of_DEGs.jpg"),width = 1024, height = 768,type="cairo")

heatmap.2 (mat1,
           # key=T,
           #keysize=1,
           # dendrogram control
           Rowv = Rowv,
           Colv=T, # if TRUE dendrogram = "both"
           distfun = dist,
           hclustfun = hclust,
           dendrogram = "both",
           symm = FALSE,
           # data scaling
           scale = "row",
           na.rm=TRUE,
           # colors
           col=colors,
           ColSideColors=sidecols,
           # level trace
           trace="none",
           density.info="none",
           #Row/Col labelling
           labRow = F,
           labCol = rld1$Sample.ReplicateGroup,
           cexCol=0.7,
           #cexRow=0.8,
           mar=c(9,11),
           srtCol=45, 
           #adjCol = c(0,1),
           # main=list("genes_changing_in_MDS_vs_healthy",cex=1.5)
)
legend("topright", legend=(levels(rld1$treatment)),
       fill= brewer.pal( 8,"Set1"), border = F,bty="n", y.intersp = 0.9, cex=0.8)
dev.off()

##############################################
## Mean TPM data table per condition #
##############################################

#### In order to highlight genes whose meanTPM values are above a certain value
### generate table with mean TPM values for each condition
### need to get sample.names 
MeanTPM_byGroup <-  cbind (
  apply(tpmdata[,rownames(colData[ colData$treatment == "DS",])], 1, mean),
  apply(tpmdata[,rownames(colData[ colData$treatment == "control",])], 1, mean))
 

colnames(MeanTPM_byGroup) <- c( "meanTPM_DS",
                               "meanTPM_control")

MeanTPM_byGroup_sigdifgenes <- MeanTPM_byGroup[rownames(MeanTPM_byGroup) %in% sigdifgenes,]
MeanTPM_byGroup_sigdifgenes <- as.data.frame(MeanTPM_byGroup_sigdifgenes)


#############################################################
#### use this to visualise genes on the list above  #########
#############################################################
# To plot TPM values 

geneplot <- function (g) {
if (grepl("^ENSG", g)){ 
  gene <- plotCounts(dds, 
                     g,  
                     intgroup = c("treatment"),
                     returnData = T)
  # obtain tpm values
  # make sure that the tpmdata and rownames(gene)  are the same
  #rownames(gene) == colnames(tpmdata)
  #gene$count <- t(tpmdata[rownames(tpmdata) == gen, ])
  # plot data
  ggplot(gene,
         aes(x = treatment, y = as.vector(count), fill = treatment, group = treatment)) + 
    geom_boxplot() + geom_point() +
    theme(axis.text.x = element_text(angle = 45, hjust = 1))+
    labs(title = lookup[lookup$ensembl_gene_id == g, 2])
} else {
  # get ENSMUSG id
  gg <- lookup[lookup$external_gene_name == g ,1]
  # get counts
  gene <- plotCounts(dds, 
                     gg,  
                     intgroup = c("treatment"),
                     returnData = T)
  # plot data
  ggplot(gene,
         aes(x = treatment, y = as.vector(count), fill = treatment, group = treatment)) + 
    geom_boxplot() + geom_point() +
    theme(axis.text.x = element_text(angle = 45, hjust = 1))+
    labs(title = g)
}
}
#save the plot
#if (grepl("^ENSMUSG", gen)){
#  ggsave( paste0(lookup[lookup$ensembl_gene_id == gen, 2],".png"))
#}  else {
#  ggsave( paste0( gen,".png"))    
#}

sink( paste0( DESeq.Dir, "SessionInfo.txt"))
sessionInfo()
sink()
###########################################################################################
##############################################################################################################
# Eva wants me to run GSEA removing genes from the duplicated region 
# in order to get a cleaner output
# I have warned him that it will be a reduction of 150 genes in a total of >10000 but he's insisted
"%w/o%" <- function(x, y) x[!x %in% y] #--  x without y
## Remove Dp1Tyb genes from rnk files
rnk_files <- ls(pattern="rnk_")
for (r in rnk_files) {
  print(r)
  rnk <- assign(r, get(r))
  ## remove genes from Dp1tyb: Dp1Tyb_genes
  rnk<- rnk[!(rnk$gene_id %in% genes_in_chr21$Row.names),]
  assign( paste0(r,"_filt"),rnk)
  write.table(rnk,  paste0(projectDir,PI, scientist, project, "/results/GSEA/", r, "_no_Hs21Genes.rnk"),
              sep="\t", quote=F, row.names = F)
}

###########################################################################################
##############################################################################################################
## Eva wants to generate heatmaps

genes.for.heatmap <- c(  "COX7C", "OGDH","ACO2",
                         "OXA1L", "HADHA","ECI1",
                         "COX7A2L", "MRPL15","SDHA",
                         "NDUFA1", "IDH3B", "LDHB", "ABCB7")

genes.for.heatmap <- merge( genes.for.heatmap, lookup, by.x=1, by.y=2)

## reorder cols by treatment
cols <- c( colData[colData$treatment == "control",1],
           colData[colData$treatment == "DS",1])

colors <- colorRampPalette( rev(brewer.pal(9, "PuOr")) )(255)

cbp <- c( "#E8ECFB", "#D9CCE3", "#CAACCB", "#BA8DB4", "#AA6F9E", "#994F88", "#882E72",
          "#1965B0", "#437DBF", "#6195CF", "#7BAFDE", "#4EB265", "#90C987", "#CAE0AB",
          "#F7F056", "#F7CB45", "#F4A736", "#EE8026", "#E65518", "#DC050C", "#A5170E",
          "#72190E", "#42150A", "#777777")

sideCols <- brewer.pal(8,"Paired")
#sidecols <- sideCols[factor(rld1$treatment)]
matf1 <- assay(rld1)[ rownames(assay(rld1)) %in% genes.for.heatmap$ensembl_gene_id, ]
matf1 <- matf1[ ,cols]
sidecols <- sideCols[factor(colData[cols,3]) ]


#colnames(matf1) <- rld1$treatment
Rowv   <- as.dendrogram(hclust(distEisen(t(matf1)), method = "ave"));

jpeg(file.path(DESeq.Dir,"Heatmap_Hallmark_genes_DS_reordered.jpg"),width = 1024, height = 768, type="cairo")

heatmap.2 (matf1,
           # key=T,
           #keysize=1,
           # dendrogram control
           Rowv = Rowv,
           Colv=F , # if TRUE dendrogram = "both"
           distfun = dist,
           hclustfun = hclust,
           dendrogram = "row",
           symm = FALSE,
           # data scaling
           scale = "row",
           na.rm=TRUE,
           # colors
           col=colors,
           ColSideColors=sidecols,
           # level trace
           trace="none",
           density.info="none",
           #Row/Col labelling
           labRow =genes.for.heatmap[,1] ,
           labCol = F,
           cexCol=1.1,
           #cexRow=0.8,
           mar=c(7,11))
 legend("topright", legend=(levels(factor(colData[cols,3]))),
       fill= sideCols, border = F,bty="n", y.intersp = 0.9, cex=0.8)
dev.off()

###########################################################################################

#cell.cycle<- read.delim("/camp/stp/babs/working/sopenam/projects/tybulewiczv/eva.lana-elola/dyrk1a_dosage_RNASeq/docs/Cellcycle_genes_for_heatmap.txt")
cell.cycle <- c(  "CHEK1",
                  "XPO1",
                  "FBXO5",
                 "PPP1CC",
                 "KIF2C",
                 "TOP2A",
                 "SMC4",
                 "CDKN2C",
                 "CCNA2",
                 "BUB1",
                 "RAD21",
                 "REC8",
                 "FOXM1",
                 "RBL1",
                 "TMPO",
                 "CDKN1B",
                 "CCNB2",
                 "MDC1",
                 "CEP70",
                 "AURKA" )

cell.cycle <- merge(cell.cycle, lookup, by.x=1, by.y=2)


## reorder cols by treatment
cols <- c( colData[colData$treatment == "control",1],
           colData[colData$treatment == "DS",1])

colors <- colorRampPalette( rev(brewer.pal(9, "PuOr")) )(255)

cbp <- c( "#E8ECFB", "#D9CCE3", "#CAACCB", "#BA8DB4", "#AA6F9E", "#994F88", "#882E72",
          "#1965B0", "#437DBF", "#6195CF", "#7BAFDE", "#4EB265", "#90C987", "#CAE0AB",
          "#F7F056", "#F7CB45", "#F4A736", "#EE8026", "#E65518", "#DC050C", "#A5170E",
          "#72190E", "#42150A", "#777777")

sideCols <- brewer.pal(8,"Paired")
#sidecols <- sideCols[factor(rld1$treatment)]
matf1 <- assay(rld1)[ rownames(assay(rld1)) %in% cell.cycle$ensembl_gene_id, ]
matf1 <- matf1[ ,cols]
sidecols <- sideCols[factor(colData[cols,3]) ]

# one gene is not expressed in this data set

#matf1 <- matf1[ -which(rowSums(matf1) == 0) ,]

#colnames(matf1) <- rld1$treatment
Rowv   <- as.dendrogram(hclust(distEisen(t(matf1)), method = "ave"));

jpeg(file.path(DESeq.Dir,"Heatmap_CellCyclegenes20_DS_reordered.jpg"),width = 1024, height = 768, type="cairo")

heatmap.2 (matf1,
           # key=T,
           #keysize=1,
           # dendrogram control
           Rowv = Rowv,
           Colv=F , # if TRUE dendrogram = "both"
           distfun = dist,
           hclustfun = hclust,
           dendrogram = "row",
           symm = FALSE,
           # data scaling
           scale = "row",
           na.rm=TRUE,
           # colors
           col=colors,
           ColSideColors=sidecols,
           # level trace
           trace="none",
           density.info="none",
           #Row/Col labelling
           labRow =cell.cycle[,1] ,
           labCol = F,
           cexCol=1.1,
           #cexRow=0.8,
           mar=c(7,11))
legend("topright", legend=(levels(factor(colData[cols,3]))),
       fill= sideCols, border = F,bty="n", y.intersp = 0.9, cex=0.8)
dev.off()


###########################################################################################

#ox.phos <- read.delim("/camp/stp/babs/working/sopenam/projects/tybulewiczv/eva.lana-elola/dyrk1a_dosage_RNASeq/docs/OxPhos_genes_for_heatmap.txt")
ox.phos <- c("ABCB7",
             "COX7C",
             "NDUFA1",
"COX15",
"NDUFB2",
"SDHB",
"IDH3G",
"COX10",
"IDH2",
"PHB2",
"CYC1",
"UQCRC2",
"AIFM1",
"NDUFS7",
"VDAC2",
"OXA1L",
"CS",
"ACO2",
"NDUFS2",
"SDHA")
ox.phos <- merge(ox.phos, lookup, by.x=1, by.y=2)


## reorder cols by treatment
cols <- c( colData[colData$treatment == "control",1],
           colData[colData$treatment == "DS",1])

colors <- colorRampPalette( rev(brewer.pal(9, "PuOr")) )(255)

cbp <- c( "#E8ECFB", "#D9CCE3", "#CAACCB", "#BA8DB4", "#AA6F9E", "#994F88", "#882E72",
          "#1965B0", "#437DBF", "#6195CF", "#7BAFDE", "#4EB265", "#90C987", "#CAE0AB",
          "#F7F056", "#F7CB45", "#F4A736", "#EE8026", "#E65518", "#DC050C", "#A5170E",
          "#72190E", "#42150A", "#777777")

sideCols <- brewer.pal(8,"Paired")
#sidecols <- sideCols[factor(rld1$treatment)]
matf1 <- assay(rld1)[ rownames(assay(rld1)) %in% ox.phos$ensembl_gene_id, ]
matf1 <- matf1[ ,cols]
sidecols <- sideCols[factor(colData[cols,3]) ]

#colnames(matf1) <- rld1$treatment
Rowv   <- as.dendrogram(hclust(distEisen(t(matf1)), method = "ave"));

jpeg(file.path(DESeq.Dir,"Heatmap_OxPhosgenes20_DS_reordered.jpg"),width = 1024, height = 768, type="cairo")

heatmap.2 (matf1,
           # key=T,
           #keysize=1,
           # dendrogram control
           Rowv = Rowv,
           Colv=F , # if TRUE dendrogram = "both"
           distfun = dist,
           hclustfun = hclust,
           dendrogram = "row",
           symm = FALSE,
           # data scaling
           scale = "row",
           na.rm=TRUE,
           # colors
           col=colors,
           ColSideColors=sidecols,
           # level trace
           trace="none",
           density.info="none",
           #Row/Col labelling
           labRow =ox.phos[,1] ,
           labCol = F,
           cexCol=1.1,
           #cexRow=0.8,
           mar=c(7,11))
legend("topright", legend=(levels(factor(colData[cols,3]))),
       fill= sideCols, border = F,bty="n", y.intersp = 0.9, cex=0.8)
dev.off()


###########################################################################################
save.image( paste0(DESeq.Dir, "DESeq_analysis.RData"))




